aiida/orm/implementation/entities.py

```diff
@@ -0,0 +1,354 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Classes and methods for backend non-specific entities"""
+
+import abc
+
+from aiida.orm.utils.node import clean_value, validate_attribute_extra_key
+
+
+class AttributesBackendEntity(abc.ABC):
```

Comment:
I would be tempted to name this differently. First of all, I think it is important to suffix it with `Mixin`. This suggests that the class is not supposed to be used on its own, but can only be functional when it is mixed in with another class. So I would suggest, `BackendEntityAttributesMixin`. Same for `BackendEntityExtrasMixin`.